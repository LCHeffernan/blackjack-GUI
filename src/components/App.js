import React, { useState } from "react";
import Player from "./Player";
import NewGame from "./NewGame";
import PlayButtons from "./PlayButtons";
import Deck from "../objects/Deck";
import Dealer from "../objects/Dealer";
import Hand from "../objects/Hand";
import "../styles/app.css";

const App = () => {
  const [playerHand, setPlayerHand] = useState({
    isHandValid: true,
    isGameOver: true,
  });
  const [handObject, setHandObject] = React.useState({});

  const handleNewGame = () => {
    const deck = new Deck();
    deck.initiateDeck();
    const dealer = new Dealer(deck);
    const hand = new Hand(dealer);
    hand.hitMe();
    hand.hitMe();

    setHandObject(hand);
    setPlayerHand({
      playerHand: hand.playerHand,
      playerScore: hand.playerScore,
      isHandValid: hand.isHandValid,
      isGameOver: hand.isGameOver,
    });
  };

  const handleHitMe = () => {
    handObject.hitMe();
    setPlayerHand({
      playerHand: handObject.playerHand,
      playerScore: handObject.playerScore,
      isHandValid: handObject.isHandValid,
      isGameOver: handObject.isGameOver,
    });
  };

  const handleStand = () => {
    handObject.stand();
    setPlayerHand({
      ...playerHand,
      isGameOver: handObject.isGameOver,
    });
  };

  return (
    <div className="app">
      <div className="table">
        <div className="title" style={{ whiteSpace: "pre" }}>
          {`♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️
♠️ ♥️ BLACKJACK TABLE ♣️ ♦️
♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️`}
        </div>
        {playerHand && (
          <Player
            className="player"
            playerHand={playerHand.playerHand}
            playerScore={playerHand.playerScore}
          />
        )}
        <div className="buttons">
          <NewGame
            handleNewGame={handleNewGame}
            isGameOver={playerHand.isGameOver}
          />
          <PlayButtons
            handleHitMe={handleHitMe}
            handleStand={handleStand}
            isGameOver={playerHand.isGameOver}
          />
        </div>
      </div>
    </div>
  );
};

export default App;
