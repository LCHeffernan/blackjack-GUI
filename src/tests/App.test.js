import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  const regexExp =
    /(\u00a9|\u00ae|[\u2000-\u3300]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff])/i;
  it("renders correctly", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct title and message", () => {
    render(<App />);

    expect(
      screen.getByText(
        "♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️ ♠️ ♥️ BLACKJACK TABLE ♣️ ♦️ ♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️"
      )
    ).toBeInTheDocument();
    expect(screen.getAllByText(regexExp).length).toEqual(1);
  });

  it("Stand button is clicked", () => {
    render(<App />);
    const newGameButton = screen.getByText("New Game");
    const standButton = screen.getByText("Stand");

    fireEvent.click(newGameButton);
    fireEvent.click(standButton);
    expect(screen.getByText(/Dealer score:/i)).toBeInTheDocument();
  });
});
