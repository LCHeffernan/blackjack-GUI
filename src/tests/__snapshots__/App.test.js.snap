// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="table"
    >
      <div
        class="title"
        style="white-space: pre;"
      >
        ♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️
♠️ ♥️ BLACKJACK TABLE ♣️ ♦️
♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️ ♥️ ♣️ ♦️ ♠️
      </div>
      <div
        class="cards-container player"
        data-testid="player-cards"
      >
        <div
          class="cards"
        >
          <div
            class="score"
          >
            Player score: 0
          </div>
          <div
            class="player-hand"
          />
        </div>
      </div>
      <div
        class="buttons"
      >
        <div>
          <button
            class="new-game-button"
            type="button"
          >
            New Game
          </button>
        </div>
        <div>
          <button
            class="play-button"
            disabled=""
            type="button"
          >
            Hit
          </button>
          <button
            class="play-button"
            disabled=""
            type="button"
          >
            Stand
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
